packages/angular_devkit/build_angular/src/builders/dev-server/vite-server.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type { BuilderContext } from '@angular-devkit/architect';
+import type { json } from '@angular-devkit/core';
+import assert from 'node:assert';
+import path from 'node:path';
+import { InlineConfig, ViteDevServer, createServer, normalizePath } from 'vite';
+import { buildEsbuildBrowser } from '../browser-esbuild';
+import type { Schema as BrowserBuilderOptions } from '../browser-esbuild/schema';
+import { loadProxyConfiguration } from './load-proxy-config';
+import type { NormalizedDevServerOptions } from './options';
+import type { DevServerBuilderOutput } from './webpack-server';
+
+export async function* serveWithVite(
+  serverOptions: NormalizedDevServerOptions,
+  builderName: string,
+  context: BuilderContext,
+): AsyncIterableIterator<DevServerBuilderOutput> {
+  // Get the browser configuration from the target name.
+  const rawBrowserOptions = (await context.getTargetOptions(
+    serverOptions.browserTarget,
+  )) as json.JsonObject & BrowserBuilderOptions;
+
+  const browserOptions = (await context.validateOptions(
+    {
+      ...rawBrowserOptions,
+      watch: serverOptions.watch,
+      poll: serverOptions.poll,
+      verbose: serverOptions.verbose,
+    } as json.JsonObject & BrowserBuilderOptions,
+    builderName,
+  )) as json.JsonObject & BrowserBuilderOptions;
+
+  let server: ViteDevServer | undefined;
+  const outputFiles = new Map<string, string>();
+  const assets = new Map<string, string>();
+  // TODO: Switch this to an architect schedule call when infrastructure settings are supported
+  for await (const result of buildEsbuildBrowser(browserOptions, context, { write: false })) {
+    assert(result.outputFiles, 'Builder did not provide result files.');
+
+    for (const file of result.outputFiles) {
+      outputFiles.set('/' + normalizePath(file.path), file.text);
```

Comment:
if one of the `outputFiles` is no longer present, will it be removed from the map anyhow?